packages/app-store/zohocalendar/lib/CalendarService.ts

```diff
@@ -0,0 +1,376 @@
+    try {
+      return {
+        ...eventRespData.events[0],
+        uid: eventRespData.events[0].uid as string,
+        id: eventRespData.events[0].uid as string,
+        type: "zoho_calendar",
+        password: "",
+        url: "",
+        additionalInfo: {},
+      };
+    } catch (error) {
+      this.log.error(error);
+      await this.deleteEvent(eventId, event, calendarId);
+      throw error;
+    }
+  }
+
+  /**
+   * @param uid
+   * @param event
+   * @returns
+   */
+  async updateEvent(uid: string, event: CalendarEvent, externalCalendarId?: string) {
+    const eventId = uid;
+    let eventRespData;
+    const calendarId = externalCalendarId || event.destinationCalendar?.externalId;
+    if (!calendarId) {
+      this.log.error("no calendar id provided in updateEvent");
+      throw new Error("no calendar id provided in updateEvent");
+    }
+    try {
+      // needed to fetch etag
+      let existingEventResponse = await this.fetcher(`/calendars/${calendarId}/events/${uid}`);
+      let existingEventData = await this.handleData(existingEventResponse, this.log);
+
+      const query = stringify({
+        eventdata: JSON.stringify({
+          ...this.translateEvent(event),
+          etag: existingEventData.events[0].etag,
+        }),
+      });
+
+      const eventResponse = await this.fetcher(`/calendars/${calendarId}/events/${uid}?${query}`, {
+        method: "PUT",
+      });
+      eventRespData = await this.handleData(eventResponse, this.log);
+    } catch (error) {
+      this.log.error(error);
+      throw error;
+    }
+
+    try {
+      return {
+        ...eventRespData.events[0],
+        uid: eventRespData.events[0].uid as string,
+        id: eventRespData.events[0].uid as string,
+        type: "zoho_calendar",
+        password: "",
+        url: "",
+        additionalInfo: {},
+      };
+    } catch (error) {
+      this.log.error(error);
+      await this.deleteEvent(eventId, event);
+      throw error;
+    }
+  }
+
+  /**
+   * @param uid
+   * @param event
+   * @returns
+   */
+  async deleteEvent(uid: string, event: CalendarEvent, externalCalendarId?: string) {
+    const calendarId = externalCalendarId || event.destinationCalendar?.externalId;
+    if (!calendarId) {
+      this.log.error("no calendar id provided in deleteEvent");
+      throw new Error("no calendar id provided in deleteEvent");
+    }
+    try {
+      // needed to fetch etag
+      let existingEventResponse = await this.fetcher(`/calendars/${calendarId}/events/${uid}`);
+      let existingEventData = await this.handleData(existingEventResponse, this.log);
+      
+      const response = await this.fetcher(`/calendars/${calendarId}/events/${uid}`, {
+        method: "DELETE",
+        headers: {
+          etag: existingEventData.events[0].etag,
+        },
+      });
+      await this.handleData(response, this.log);
+    } catch (error) {
+      this.log.error(error);
+      throw error;
+    }
+  }
+
+  async getAvailability(
```

Comment:
When creating events in the calendar, those slots are not being blocked.

Reply:
I have just checked, when I create events in the calendar, he slots get actually blocked.

![image](https://github.com/calcom/cal.com/assets/25750396/657d216d-07e7-4753-b78c-601a3b6fb7c1)
![image](https://github.com/calcom/cal.com/assets/25750396/4b3b4e73-29dd-4a93-931a-3a0d35f68d85)

Reply:
Are there specific settings that I have to enable in Zoho? This would also be good to communicate with other users.

![CleanShot 2023-08-21 at 15 00 18](https://github.com/calcom/cal.com/assets/65426560/e2b0bb12-e246-49b3-b7c9-cac9434cac6c)

![CleanShot 2023-08-21 at 15 01 14](https://github.com/calcom/cal.com/assets/65426560/53b04e7e-706e-4605-b9d2-0d23f20dafed)

Reply:
So I double checked and while some slots are being blocked, it's not during the correct times. I have these two events in my calendar that are under "America/Toronto"
![CleanShot 2023-08-21 at 15 14 53](https://github.com/calcom/cal.com/assets/65426560/4277cdf3-a3db-4792-8eae-0ac94d2db21e)

Although when querying for free/busy times and then converting them to `busyData` the times are not lining up.
![CleanShot 2023-08-21 at 15 16 07](https://github.com/calcom/cal.com/assets/65426560/be471e5b-db50-4fc0-a0bf-51cbc1828bbf)

Reply:
Fixed!
![image](https://github.com/calcom/cal.com/assets/25750396/68df51d1-5a3d-46c7-8ef0-58e3b3b0ee98)